add: extra colorschemes for ghostty, alacritty, kitty and foot

skewb1k opened the pull request:
Hi!
I added support for several terminal emulators: Ghostty, Alacritty, Kitty, and Foot. The new colorscheme files are under the extras/ directory as was suggested in #25

There is a screenshot of foot, in other terminals looks about the same.
![2025-04-30-044902_sway-screenshot](https://github.com/user-attachments/assets/2a91d5ec-2a5c-499b-ae3e-62e0eaf2d00b)

Let me know what you think about it — happy to adjust anything if needed!

vague2k commented:
Looks great! 

If it's possible, I'd love to see what it would look like with less red/warm colors. That would be neat!

skewb1k commented:
Yeah, i agree that it's maybe too warm because of red(func) color is one of the primary ones. But the reason for that i just aligned it with rose pine.
![2025-05-01-201449_sway-screenshot](https://github.com/user-attachments/assets/593793fe-be33-4a3d-a843-4346a2e16ce3)
![2025-05-01-203121_sway-screenshot](https://github.com/user-attachments/assets/955b0dfa-70d9-4b9d-a4b3-6515dcc5490e)

And it's obvious but maybe not the best solution, because red is too dimmed. So tried to swap it with other primary colors. 
![2025-05-01-204030_sway-screenshot](https://github.com/user-attachments/assets/ef48abb9-4513-4870-aa2e-411086960d3c)
![2025-05-01-201931_sway-screenshot](https://github.com/user-attachments/assets/df8cbcb6-6b3c-4da6-b065-7dfabddfc1eb)
![2025-05-01-202151_sway-screenshot](https://github.com/user-attachments/assets/0f8d2319-f85d-453b-b349-0694c95812f4)

I'm not sure that any of this is better. So maybe do not use red completely, here is version with replaced light blue(builtin)
![2025-05-01-203552_sway-screenshot](https://github.com/user-attachments/assets/9610e432-7ca6-462e-ae97-0b1c8816dd57)

As a result i'm a bit confused what is best, maybe i picked the wrong base colors. I'll be glad to hear your opinion, maybe you liked something, maybe you see how colors should be mapped for terminal.

Speaking for myself i really like the nvim theme and the first "warm" port, it reminds me gruvbox-ish theming.

vague2k commented:
As far as this theme goes, I associate red/warm colors with either important info, such as function names and/or errors.

I think out of the different variations you've shown, I think I like the last screenshot the best. where you replaced the red with light blue(builtin). I think this is more accurate as to what i envisioned in my head.

skewb1k commented:
the last option looks nice, but I found that 4 and 6 colors are too similar and it can cause problems, for example eza symlinks highlighting not works (i think it applies for some other tools as well)
![2025-05-02-083858_sway-screenshot](https://github.com/user-attachments/assets/df5d6381-eda1-4a9a-89fa-345bf12de49c)

So i tried some other colors as replacement for initial red. 1 its operator color, 2 - constant. Personally, i prefer the first one.
![2025-05-02-083349_sway-screenshot](https://github.com/user-attachments/assets/af3aab5c-9b53-4fd3-b0fd-6e049db238d6)
![2025-05-02-083652_sway-screenshot](https://github.com/user-attachments/assets/a306ae3f-1f49-4284-909e-f992303bfa15)

Also i remembered about the terminal mode in nvim, which currently uses green for color2, which corresponds with terminal ANSI colors or whatever that thing is called, but i think it shouldn't be used in terminal theme, because in nvim it's used only for gitdiffs. Ideally, nvim's terminal mode should match the terminal emulator's colorscheme. I can work on this and commit after we decide on the colors. As always, waiting for your thoughts on all of this!

vague2k commented:
Out of the 2 screenshots in your most recent reply, i like the 2nd one the most! I think that's it!

skewb1k commented:
I adjusted colors and fix nvim terminal mode to match them. Here is the final screenshot
![2025-05-04-134203_sway-screenshot](https://github.com/user-attachments/assets/87064b44-0cb9-4003-917e-810e3584659d)

skewb1k commented:
Will this be merged? Do I need to change anything?
I understand if you're busy or not reviewing things right now. Just checking in and happy to adjust anything if needed.

vague2k commented:
@skewb1k sorry for not getting back to you, it'll get merged I've just been busy with a new job and stuff. 

I probably have some time tonight to review things myself manually on each terminal emulator and decided if there's anything that should be changed. From what i've seen so far it looks good. I'll let you know regardless

vague2k commented:
I changed some of the coloring to resemble a more traditional looking terminal colorscheme. I also corrected the `plus` and `error` colors to be just a tad darker and saturated.

I used a modified version of afterglow (dark bg) on a day to day, I tried to make the extras palette match this scheme as closely as possible without deviating from vague's color palette
<img width="671" alt="Screenshot 2025-05-14 at 9 20 42 PM" src="https://github.com/user-attachments/assets/f286441e-42dc-43e4-b60d-b4793f445c1f" />

<img width="672" alt="Screenshot 2025-05-14 at 10 03 42 PM" src="https://github.com/user-attachments/assets/4e173d88-bb39-4392-af21-dc439e25d86e" />

Feel free to use it and let me know if it's still what you were expecting! I personally think this is even closer to what I envisioned

skewb1k commented:
i like the result pretty far. I realized that green is indeed a better and more idiomatic choice in the context of terminal colors. I just found this kinda weird that it is used in a lot of places in terminal while in nvim theme its only for git diffs, but for example for zsh highlighting i just changed it to other color and all seems fine.

I fixed the foot config—it was invalid (I assume foot is Wayland-only and you haven’t tested it). I also adjusted the README to reflect the correct plus and error colors.

Also you committed some leading whitespaces, checkout [mini.trailspace](https://github.com/echasnovski/mini.trailspace) which I recently added support for in this theme. It highlights it and has ability to trim them on save, which is useful for files where formatters are unavailable, like this:
```lua
vim.api.nvim_create_autocmd('BufWritePre', {
    group = vim.api.nvim_create_augroup('trailspace', { clear = true }),
    callback = require('mini.trailspace').trim,
})
```

skewb1k commented:
i'm not sure about adding bright colors: on the one hand I haven't noticed them being used often, but on the other hand it seems better to have them different from the standard ones anyway. For the first versions I added them by raising the lightness in hsl by 20 or something. What do you think?

skewb1k commented:
done. But i didn't touch nvim builtin terminal: not sure where to define colors. In internals DEFAULT_SETTINGS? Or just hardcode them in terminal.lua since they are only for terminal mode?

vague2k commented:
i'm pretty sure hardcoding them is perfectly fine for now since these bright colors wouldn't be used anywhere else (at least for now).

vague2k commented:
@skewb1k I think this pr is fit to merge now. I'm probably gonna release a new version for the colorscheme while I'm at it.

Thanks for liking the colorscheme so much that you went out of your way to make a contribution! i love how everything ended up coming out
